static/dream_web/index.js

```diff
@@ -53,14 +53,22 @@ function clearFields(form) {
     form.prompt.value = prompt;
 }
 
+const BLANK_IMAGE_URL = 'data:image/svg+xml,<svg xmlns="http://www.w3.org/2000/svg"/>';
 async function generateSubmit(form) {
     const prompt = document.querySelector("#prompt").value;
 
     // Convert file data to base64
     let formData = Object.fromEntries(new FormData(form));
     formData.initimg = formData.initimg.name !== '' ? await toBase64(formData.initimg) : null;
 
-    document.querySelector('progress').setAttribute('max', formData.steps);
+    let progressSectionEle = document.querySelector('#progress-section');
+    progressSectionEle.style.display = 'initial';
+    let progressEle = document.querySelector('#progress-bar');
+    progressEle.setAttribute('max', formData.steps);
+    let progressImageEle = document.querySelector('#progress-image');
+    progressImageEle.src = BLANK_IMAGE_URL;
+
+    progressImageEle.style.display = {}.hasOwnProperty.call(formData, 'progress_images') ? 'initial': 'none';
```

Comment:
Why not just do `formData.hasOwnProperty`?

Reply:
You should never do `foo.hasOwnProperty`. That [will break](https://stackoverflow.com/a/12017703) if `foo` has an own property named `hasOwnProperty`, or if it has a null prototype. In this particular instance we know it won't, but this is still the idiomatic way to write this test.